index.js

```diff
@@ -181,3 +185,55 @@ AddonManager.addInstallListener({
     app.send('addon-install:download-failed', formatInstallData(install));
   }
 });
+
+function pingServer(title, addon) {
+  cookieManager.getCookiesFromHost('ideatown.dev', function(cookies) {
+    let csrf;
+    let cookieStr = '';
+    cookies.forEach(function(c) {
+      cookieStr += c.name + '=' + c.value + ' ';
+      if (c.name === 'csrftoken') csrf = c.value;
+    });
+
+    if (csrf) {
+      getMe(csrf, cookieStr, function(id) {
+        request({
+          url: prefs.METRICS_URL,
+          content: formatEvent(title, id, addon),
+          headers: {
+            'Cookie': cookieStr,
+            'X-CSRFToken': csrf
+          },
+          onComplete: function(response) {
+            console.error(response.text);
+          }
+        }).post();
+      });
+    } else app.send('addon-self:needs-auth');
+  });
+}
+
+function getMe(csrf, cookieStr, cb) {
+  request({
+    url: prefs.USER_URL + '?format=json',
+    headers: {
+      'Cookie': cookieStr,
+      'X-CSRFToken': csrf
+    },
+    onComplete: function(response) {
+      cb(response.json.id);
```

Comment:
Maybe replace with `cb(response.json && response.json.id)`, to avoid throwing in the case that `response.json` is undefined--unless `request` guarantees that `response.json` will not be undefined (does it?)